src/main/kotlin/io/zerobase/smarttracing/resources/DevicesResource.kt

```diff
@@ -1,34 +1,70 @@
 package io.zerobase.smarttracing.resources
 
-import io.zerobase.smarttracing.models.IdWrapper
+import com.fasterxml.jackson.annotation.JsonProperty
+import io.zerobase.smarttracing.GraphDao
+import io.zerobase.smarttracing.models.*
 import javax.ws.rs.*
 import javax.ws.rs.core.MediaType
 
 //region Request Models
+data class CreateDeviceRequest(@JsonProperty("fingerprint") val fingerprint: Fingerprint)
+
+data class CreateCheckInRequest(
+        @JsonProperty("scannedId") val scannedId: ScannableId,
+        @JsonProperty("type") val type: ScanType,
+        @JsonProperty("location") val location: Location?
+) {
+    enum class ScanType {
+        DEVICE_TO_DEVICE,
+        DEVICE_TO_SCANNABLE
+    }
+}
+
+data class UpdateCheckInLocationRequest(
+        @JsonProperty("lat") val lat: Float,
+        @JsonProperty("long") val long: Float
+)
 
 //endregion
 
 @Path("/devices")
 @Consumes(MediaType.APPLICATION_JSON)
 @Produces(MediaType.APPLICATION_JSON)
-class DevicesResource {
+class DevicesResource(val dao: GraphDao) {
 
     @POST
     @Creator
-    fun createDevice(request: Any): IdWrapper {
-        TODO("not implemented")
+    fun createDevice(request: CreateDeviceRequest): IdWrapper? {
+        val fingerprint = request.fingerprint
+
+        val newDeviceId = dao.createDevice(fingerprint, null)
+
+        return newDeviceId?.let { IdWrapper(newDeviceId) }
     }
 
-    @Path("/{id}/check-ins")
+    @Path("/{deviceId}/check-ins")
```

Comment:
nit: the context of the id is implied by the scope of the class

Reply:
Fixed